MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -4309,37 +4362,33 @@ public void setOriginalUnit(Unit unit) {
     }
 
     public int getNumShares(boolean sharesForAll) {
```

Comment:
I think this method is going to require an explanation of the sharesForAll parameter. Also, I'm wondering if it'll be a little more efficient if we cache the results of all this logic. No need to store it permanently, just invalidate it when something happens to change the number of shares.

Reply:
Oh, it will be... but that's for a future PR (trying to keep features separate, and that is definitely a new internal feature). As for the comment, that has been added.